### PR TITLE
9858 + 3898 + 9874

### PR DESCRIPTION
**Pull request by TomElliottFlexion**
Functionality to test

### CODE REVIEW

File: shared/src/business/useCaseHelper/caseAssociation/associateIrsPractitionerToCase.ts [hunk 1]
```diff
@@ -14,46 +13,57 @@ const { UserCase } = require('../../entities/UserCase');
  */
 exports.associateIrsPractitionerToCase = async ({
   applicationContext,
+  consolidatedCasesDocketNumbers = [],
   docketNumber,
   serviceIndicator,
   user,
 }) => {
```
Comment (TomElliottFlexion) [inline shared/src/business/useCaseHelper/caseAssociation/associateIrsPractitionerToCase.ts:25]: Consider typing this

File: web-client/src/presenter/actions/FileDocument/setIsExternalConsolidatedCaseGroupEnabledValueAction.ts [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+  const { overrideIsMultiDocketableEventCode } = props;
+
+  const isConsolidatedGroupAccessEnabled = get(
+    state.featureFlags[
+      ALLOWLIST_FEATURE_FLAGS.CONSOLIDATED_CASES_GROUP_ACCESS_PETITIONER.key
+    ],
+  );
+  const isInConsolidatedGroup = !!caseDetail.leadDocketNumber;
+  const isMultiDocketableEventCode =
+    overrideIsMultiDocketableEventCode ??
```
Comment (TomElliottFlexion) [inline web-client/src/presenter/actions/FileDocument/setIsExternalConsolidatedCaseGroupEnabledValueAction.ts]: Consider renaming to overrideForRequestAccess?

File: web-client/src/presenter/computeds/externalConsolidatedCaseGroupHelper.ts [hunk 3]
```diff
@@ -0,0 +1,61 @@
...
+    const roleToDisplay = party => {
+      if (party.role === USER_ROLES.privatePractitioner) {
+        return 'Petitioner Counsel';
+      } else if (party.role === USER_ROLES.irsPractitioner) {
+        return 'Respondent Counsel';
+      } else {
+        return CONTACT_TYPE_TITLES[party.contactType];
+      }
+    };
+    // simplify with lodash groupBy function
```
Comment (TomElliottFlexion) [inline web-client/src/presenter/computeds/externalConsolidatedCaseGroupHelper.ts]: ?

File: web-client/src/presenter/sequences/gotoFileDocumentSequence.ts [hunk 4]
```diff
@@ -29,6 +31,9 @@ const gotoFileDocument = startWebSocketConnectionSequenceDecorator([
   ),
   getCaseAction,
   setCaseAction,
+  //TODO: put under feature flag
```
Comment (TomElliottFlexion) [inline web-client/src/presenter/sequences/gotoFileDocumentSequence.ts]: !

File: shared/src/business/useCaseHelper/caseAssociation/associateIrsPractitionerToCase.ts (hunk 1 above)
Comment (shumway-tm) [reply shared/src/business/useCaseHelper/caseAssociation/associateIrsPractitionerToCase.ts:25]: resolved

File: web-client/src/presenter/actions/FileDocument/setIsExternalConsolidatedCaseGroupEnabledValueAction.ts (hunk 2 above)
Comment (shumway-tm) [reply web-client/src/presenter/actions/FileDocument/setIsExternalConsolidatedCaseGroupEnabledValueAction.ts]: resolved

File: web-client/src/presenter/computeds/externalConsolidatedCaseGroupHelper.ts (hunk 3 above)
Comment (shumway-tm) [reply web-client/src/presenter/computeds/externalConsolidatedCaseGroupHelper.ts]: resolved.

File: web-client/src/presenter/sequences/gotoFileDocumentSequence.ts (hunk 4 above)
Comment (shumway-tm) [reply web-client/src/presenter/sequences/gotoFileDocumentSequence.ts]: resolved

**Review by shumway-tm — APPROVED** [review]

**Pull request merged**